aether-ui/aether/ui/api/views.py

```diff
@@ -53,16 +53,22 @@ def fetch(self, request, *args, **kwargs):
     @action(detail=True, methods=['post'], url_path='delete-artefacts')
     def delete_artefacts(self, request, pk=None, *args, **kwargs):
         pipeline = self.get_object_or_404(pk=pk)
-        data = request.data
+        artefacts_result = {'not_published': True}
         if pipeline.mappingset:
-            return utils.delete_operation(
-                f'mappingsets/{pipeline.mappingset}/delete-artefacts/',
-                data,
-                pipeline
-            )
+            try:
+                artefacts_result = utils.delete_operation(
+                    f'mappingsets/{pipeline.mappingset}/delete-artefacts/',
+                    request.data,
+                    pipeline,
+                )
+                if not artefacts_result:
```

Comment:
this `if` can go outside, can't it? :thinking: 
```python
if not artefacts_result:
    return Response(status=status.HTTP_204_NO_CONTENT)
return Response(artefacts_result)
```

Reply:
Yes it can, but I rather exit as soon as I can and avoid checking `if not artefacts_result` on error requests

Reply:
To be consistent maybe if the pipeline wasn't published yet the response should also be `204` :thinking

```python
    @action(detail=True, methods=['post'], url_path='delete-artefacts')
    def delete_artefacts(self, request, pk=None, *args, **kwargs):
        pipeline = self.get_object_or_404(pk=pk)

        if pipeline.mappingset:
            try:
                artefacts_result = utils.delete_operation(
                    f'mappingsets/{pipeline.mappingset}/delete-artefacts/',
                    request.data,
                    pipeline,
                )
                if artefacts_result:
                    return Response(artefacts_result)
            except Exception as e:
                return Response(str(e), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
        else:
            pipeline.delete()

        return Response(status=status.HTTP_204_NO_CONTENT)
```

Reply:
What do you think?

Reply:
The implementation is to differentiate between an unpublished pipeline and a published pipeline with a missing upstream mappingset

Reply:
would it be better in that case to return a "missing artefacts" warning? (up to you) it's fine as it is.